modules/system/applications.nix

```diff
@@ -24,12 +24,12 @@ in
       # Set up applications.
       echo "setting up ~/Applications..." >&2
 
-      if [ ! -e ~/Applications -o -L ~/Applications ]; then
-        ln -sfn ${cfg.build.applications}/Applications ~/Applications
-      elif [ ! -e ~/Applications/Nix\ Apps -o -L ~/Applications/Nix\ Apps ]; then
+      mkdir -p ~/Applications
```

Comment:
Wouldn't mkdir fail if the directory already exists? Probably better to leave the existance check in and simply mkdir instead of ln'ing.

Reply:
The magic of the `-p` flag of mkdir is that it doesn't report any error if the folder to be created already exists (as well as the hierarchy of directories)

Reply:
Oh deat, didn't know that!